.github/workflows/release.yml

```diff
@@ -0,0 +1,131 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      dryRun:
+        description: "Do a dry run to preview instead of a real release"
+        required: true
+        default: "true"
+
+jobs:
+  authorize:
+    name: Authorize
+    runs-on: macos-10.15
+    steps:
+      - name: ${{ github.actor }} permission check to do a release
+        uses: octokit/request-action@v2.0.0
+        with:
+          route: GET /repos/:repository/collaborators/${{ github.actor }}
+          repository: ${{ github.repository }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    name: Release
+    runs-on: macos-10.15
+    needs: [authorize]
+    env:
+      SWIFT_DOC_VERSION: '1.0.0-beta.6'
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set Xcode 12.4
+        run: |
+          sudo xcode-select -switch /Applications/Xcode_12.4.app
+
+      - name: iOS Tests
+        run: |
+          xcodebuild test \
+            -project Skylab.xcodeproj \
+            -scheme Skylab \
+            -sdk iphonesimulator \
+            -destination 'platform=iOS Simulator,name=iPhone 12,OS=14.4'
+
+      - name: macOS Tests
+        run: |
+          xcodebuild \
+            -project Skylab.xcodeproj \
+            -scheme Skylab \
+            -sdk macosx \
+            -destination 'platform=macosx' \
+            test
+
+      - name: tvOS Tests
+        run: |
+          xcodebuild \
+            -project Skylab.xcodeproj \
+            -scheme Skylab \
+            -sdk appletvsimulator \
+            -destination 'platform=tvOS Simulator,name=Apple TV' \
+            test
+
+      - name: Validate Podfile
+        run: pod lib lint
+
+      - name: Cache Swift Doc
+        id: cache-swift-doc
+        uses: actions/cache@v2
+        with:
+          path: /usr/local/bin/swift-doc
+          key: ${{ runner.os }}-swift-doc-${{ env.SWIFT_DOC_VERSION }}
+
+      # Swift Doc requires graphviz as a dependency. 
+      # TODO Look into either caching graphviz or using a separate linux job 
+      #      in order to use the pre-built SwiftDoc github action.
+      - name: Install Graphviz
+        run: brew install graphviz
+
+      - name: Install Swift Doc
+        if: steps.cache-swift-doc.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/SwiftDocOrg/swift-doc
+          cd swift-doc
+          git checkout ${{ env.SWIFT_DOC_VERSION }}
+          make install
+          cd ..
+          rm -rf swift-doc
+
+      - name: Generate Swift Doc
+        run: swift doc generate Sources/Skylab/ --module-name Skylab --output docs --format html --base-url /skylab-ios-client
```

Comment:
`skylab-ios-client` can probably be pulled into a variable that is easy to change based on the repo name